dap-mode.el

```diff
@@ -1138,18 +1155,20 @@ CONFIGURATION-SETTINGS - plist containing the preset settings for the configurat
         (file-error (setq success t))))
     port))
 
-(defun dap--select-template ()
-  "Select the configuration to launch."
+(defun dap--select-template (&optional origin)
+  "Select the configuration to launch.
```

Comment:
Here we talk about "configuration" in the comment but the function name is about "templates". I've proposed some terminology around DAP concepts so that we can understand and model things things more clearly.

Reply:
Yes, completely agree with you on this.
Actually, I did recommend a new command for this, i.e: dap-edit-debug-configuration in the original ticket but the concern was that it will complicate things a lot.
But maybe it is better to fix them all at the same time then we're good.

Reply:
I think it is fine to split it into several commits, so let us work on making this PR a step into what we want to achieve.

Reply:
Sure, that is the best approach.